controllers/imapmon.go

```diff
@@ -0,0 +1,184 @@
+package controllers
+
+/* TODO:
+*		 - Have a counter per config for number of consecutive login errors and backoff (e.g if supplied creds are incorrect)
+*		 - Have a DB field "last_login_error" if last login failed
+*		 - DB counter for non-campaign emails that the admin should investigate
+*		 - Add field to User for numner of non-campaign emails reported
+ */
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/gophish/gophish/logger"
+
+	"github.com/glennzw/eazye"
+	"github.com/gophish/gophish/config"
+	"github.com/gophish/gophish/models"
+)
+
+// Pattern for GoPhish emails e.g ?rid=AbC123
+var goPhishRegex = regexp.MustCompile("(\\?rid=[A-Za-z0-9]{7})")
+
+// ImapMonitor is a worker that monitors IMAP serverd for reported campaign emails
+type ImapMonitor struct {
+	cancel    func()
+	reportURL string
+	checkFreq int
+}
+
+// PeriodicIMAPMonitor will periodically check the database for
+func PeriodicIMAPMonitor(ctx context.Context, im *ImapMonitor) {
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			imaps, err := models.GetEnabledIMAPs()
+			if err != nil {
+				log.Error(err)
+				break
+			}
+			var wg sync.WaitGroup
+			wg.Add(len(imaps))
+			for _, i := range imaps {
+				//We launch a goroutine for each IMAP account. In the future should use some more sophisticated worker/queues
+				go checkForNewEmails(i, &wg, im.reportURL)
+			}
+			wg.Wait()
+			time.Sleep(time.Duration(im.checkFreq) * time.Second)
+		}
+	}
+
+}
+
+// NewImapMonitor returns a new instanfe of the ImapMonitor
+//func NewImapMonitor(configP config.PhishServer, configA *config.Config) *ImapMonitor {
+func NewImapMonitor(config *config.Config) *ImapMonitor {
+	// Make sure database connection exists. Not sure why I have to do this here, but
+	//  otherwise db is <nil> when calling models.GetEnabledIMAPs()
+	err := models.Setup(config)
```

Comment:
Hmm.... yeah, you shouldn't need to do this. Let me think on this a bit once I get the initial review done.

Reply:
We don't need the `GetEnabledIMAPs()` function with the new polling logic, so removed this and all seems fine.